keras/src/backend/tensorflow/numpy.py

```diff
@@ -35,6 +35,25 @@ def add(x1, x2):
     )
     x1 = convert_to_tensor(x1, dtype)
     x2 = convert_to_tensor(x2, dtype)
+
+    # Special case of `tf.add`: `tf.nn.bias_add`
+    # `BiasAdd` can be fused with `MatMul` and `Conv*` kernels
+    # Expecting `x1` to be `inputs` and `x2` to be `bias` (no swapping)
+    x2_squeeze_shape = [d for d in x2.shape if d is None or d > 1]
```

Comment:
We don't know what exactly user want's to add. It can be 2 input tensors with batch dimension set to None (instead of inputs + bias).
Bias should have no None's, and few lines below we will check that.

Reply:
I thought we were only support the case of adding a bias in Dense or Conv2D? In this case the bias term always has an explicit shape and there are only 3 different shapes to look for.

Reply:
Let's see:
- Conv1D
- Conv2D
- Conv3D
- Dense
- Custom layers (something like add relative positional bias to q @ k.T)

If you don't like this implicit solution, maybe it would be better to define `bias_add` in ops and use it explicitly?